vmdb/app/controllers/ems_common.rb

```diff
@@ -674,7 +674,7 @@ def set_record_vars(ems, mode = nil)
     ems.provider_region = @edit[:new][:provider_region]
     ems.hostname = @edit[:new][:hostname]
     ems.ipaddress = @edit[:new][:ipaddress]
-    ems.port = @edit[:new][:port] if ["openstack", "openstack_infra", "rhevm"].include?(ems.emstype)
+    ems.port = @edit[:new][:port] if ems.class.supports_port?
```

Comment:
Agreed ... it threw me that it was asking at the class level.  But, then reading the changes to the class, it made more sense.